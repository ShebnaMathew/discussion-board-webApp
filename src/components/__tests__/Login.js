import Login from "../Login";
import renderer from "react-test-renderer";
import {render, fireEvent, screen} from "@testing-library/react";


// DOM testing

test("Logging in renders the Home and clicking Log out, logs out", () => {
    const mockSubmit = jest.fn();

    render(<button type="button" className="btn btn-secondary float-end" onClick={mockSubmit}>
    Log in
    </button>)
    fireEvent.click(screen.getByRole("button"));

    expect(mockSubmit).toHaveBeenCalled();

    render(<button type="button" className="btn btn-secondary float-end" onClick={mockSubmit}>
    Log Out
    </button>)

expect(mockSubmit).toHaveBeenCalled();

})





