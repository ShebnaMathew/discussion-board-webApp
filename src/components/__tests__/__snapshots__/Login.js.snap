// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login button rendered 1`] = `
<button
  class="btn btn-outline-light"
  onClick={[MockFunction]}
  type="button"
>
  "Login"
</button>
`;
